core/types/src/keys.rs

```diff
@@ -71,6 +79,29 @@ impl<ADDR, KIND, KEY: Default + Zeroize> Default for Key<ADDR, KIND, KEY> {
     }
 }
 
+/// Expose [`ReprBytes`] for internal `KEY` types implementing this
+impl<ADDR, KIND, KEY> ReprBytes for Key<ADDR, KIND, KEY>
+where
+    KEY: ReprBytes + Default + Zeroize,
```

Comment:
IIUC this impl only requires `ReprBytes`
```suggestion
    KEY: ReprBytes,
```

Reply:
i think you're right. i don't have strong opinions / am happy to change if it's preferred?
but had it this way so that any instance of `KEY` has the same fundamental bounds (like, it's slightly duplicated perhaps but is there a reason to have more relaxed bounds on an impl than the object itself?)

Reply:
It's normal to not `Zeroize` public key types, however, if we want to err on the side of builtin safety/privacy, we probably want _all key material_ to be zeroized. If leak of a public key could give away that you were a ring input, for example, then a small write (which likely in most ISAs is going to compile down to a `xor eax eax` or similar) then it's such a small overhead that we may as well `#[derive(Zeroize)]` for `KEY`.